pkg/apis/v1beta1/nodepool.go

```diff
@@ -202,3 +210,72 @@ func (pl *NodePoolList) OrderByWeight() {
 		return ptr.Int32Value(pl.Items[a].Spec.Weight) > ptr.Int32Value(pl.Items[b].Spec.Weight)
 	})
 }
+
+// GetAllowedDisruptions returns the minimum allowed disruptions across all disruption budgets for a given node pool.
+// This returns two values as the resolved value for a percent depends on the number of current node claims.
+func (in *NodePool) GetAllowedDisruptions(ctx context.Context, c clock.Clock) (intstr.IntOrString, intstr.IntOrString, error) {
```

Comment:
Rather than handling everything as `IntOrString` does it make sense to just divide up the budgets into percentage-based ones and integer-based ones and take the minimum that way rather than handling both cases simultaneously and trying to split it later?

Reply:
I could see an argument for this. This is essentially what I do for the comparison in this function, but if we don't return an intstr.IntOrString() here, we have to create a new one in the caller function to then create the intstr.IntOrString() as an input to GetScaledValue().